src/tracker/tracker.jl

```diff
@@ -0,0 +1,57 @@
+#---------------------
+# TRACKER COMPONENTS
+#---------------------
+
+include("../core.jl")
+include("tracker_sampler.jl")
+# include("tracker_state_estimator.jl")
+# include("tracker_model.jl")
+# include("tracker_features.jl")
+
+abstract type AbstractTracker end
+mutable struct TrackerBoosting{I <: Integer, B <: BoxROI, CS <: CurrentSampler} <: AbstractTracker
+    # initialized
+    box::B
+    sampler::CS
+    num_of_classifiers::I
+    initial_iterations::I
+    num_of_features::I
+
+    # constructor
+    function TrackerBoosting{I, B, CS}(box::B, sampler::CS, num_of_classifiers::I = 100, initial_iterations::I = 20,
+        num_of_features::I = 1050)where {I <: Integer, B <: BoxROI, CS <: CurrentSampler}
+        if size(box.bound, 1) != 4
+            error("Invalid bounding box size")
+        end
+
+        if box.bound[1] < 1 && box.bound[1] > size(box.img)[1]
```

Comment:
These checks should have `OR` rather than `AND`.

Reply:
Good idea. I'll move them to `BoxROI` constructor then.